dockertest/dockercmd.py

```diff
@@ -379,6 +422,27 @@ def process_id(self):
                                   " called.")
         return self._async_job.sp.pid
 
+    @property
+    def container_id(self):
+        """
+        Try to discover our own container ID or name. Return None if we can't.
+        """
+
+        # The simple case: if we are a docker attach command, assume that
+        # subargs are zero or more flags plus a container ID or name.
+        if self.subcmd == 'attach':
+            return self.subargs[-1]
+
+        # Non-attach command. Find our PID, iterate over all containers,
+        # get their PID (via inspect). If we find a match, return the CID.
+        pid = self.process_id
+        cids = utils.run('docker ps -q', verbose=False).stdout.splitlines()
+        for cid in cids:
+            c_pid = utils.run('docker inspect --format {{.State.Pid}} ' + cid)
+            if int(c_pid.stdout) == int(pid):
+                return cid
+        return None
+
```

Comment:
Hmmmm, I don't think the CID can ever change in the context of this class (I don't remember if we allow re-use after the container exits).  Consider caching the found CID in a private instance variable.  The caller may be referencing this property inside a loop (in fact ``wait_for_ready()``) does this.  Spawning processes is "expensive" and if the system is under particularly high load, can lead to all sorts of problems (depending on how tight the loop is and how quickly the ``utils.run()`` returns).